### PR TITLE
configure bootstrap, fork resolution, pop rewards etc via the config file

### PR DESCRIPTION
**Pull request by eu321**

### CODE REVIEW

File: alt-integration-grpc-service/src/main/java/org/veriblock/webservice/DefaultConfiguration.java [hunk 1]
```diff
@@ -77,6 +91,156 @@ public NetworkParameters getVeriblockNetworkParameters() {
...
+    }
+
+    public ForkresolutionConfig getForkresolutionConfig() {
+        String keystoneFinalityDelay = properties.getProperty("Forkresolution.keystoneFinalityDelay");
+        String amnestyPeriod = properties.getProperty("Forkresolution.amnestyPeriod");
+        if (keystoneFinalityDelay == null && amnestyPeriod == null)
+            return null;
+
+        if (keystoneFinalityDelay == null || amnestyPeriod == null)
+            throw new IllegalArgumentException("Must specify either all of none of the Forkresolution config values");
```
Comment (blackyblack) [inline alt-integration-grpc-service/src/main/java/org/veriblock/webservice/DefaultConfiguration.java]: "either all or none"

File: alt-integration-grpc-service/src/main/java/org/veriblock/webservice/DefaultConfiguration.java [hunk 2]
```diff
@@ -77,6 +91,156 @@ public NetworkParameters getVeriblockNetworkParameters() {
...
+        String aboveIntendedPayoutMultiplierKeystone = properties.getProperty("PopRewardCurveConfig.aboveIntendedPayoutMultiplierKeystone");
+        if (startOfDecreasingLine == null && widthOfDecreasingLineNormal == null &&
+            widthOfDecreasingLineKeystone == null && aboveIntendedPayoutMultiplierNormal == null &&
+            aboveIntendedPayoutMultiplierKeystone == null)
+            return null;
+
+        if (startOfDecreasingLine == null || widthOfDecreasingLineNormal == null ||
+            widthOfDecreasingLineKeystone == null || aboveIntendedPayoutMultiplierNormal == null ||
+            aboveIntendedPayoutMultiplierKeystone == null)
+            throw new IllegalArgumentException("Must specify either all of none of the PopRewardCurve config values");
```
Comment (blackyblack) [inline alt-integration-grpc-service/src/main/java/org/veriblock/webservice/DefaultConfiguration.java]: "either all or none"

File: alt-integration-grpc-service/src/main/java/org/veriblock/webservice/DefaultConfiguration.java [hunk 3]
```diff
@@ -77,6 +91,156 @@ public NetworkParameters getVeriblockNetworkParameters() {
...
+            popDifficultyAveragingInterval == null && popRewardSettlementInterval == null)
+            return null;
+
+        if (basicReward == null || payoutRounds == null ||
+            keystoneRound == null || roundRatios == null ||
+            maxRewardThresholdNormal == null || maxRewardThresholdKeystone == null ||
+            flatScoreRound == null || flatScoreRoundUse == null ||
+            curveConfig == null || relativeScoreLookupTable == null ||
+            popDifficultyAveragingInterval == null || popRewardSettlementInterval == null)
+            throw new IllegalArgumentException("Must specify either all of none of the PopRewardCalculator config values");
```
Comment (blackyblack) [inline alt-integration-grpc-service/src/main/java/org/veriblock/webservice/DefaultConfiguration.java]: "either all or none"

**Review by blackyblack — APPROVED** [review]

File: alt-integration-grpc-service/src/main/java/org/veriblock/webservice/DefaultConfiguration.java [hunk 4]
```diff
@@ -77,6 +91,156 @@ public NetworkParameters getVeriblockNetworkParameters() {
...
+    }
+
+    public BitcoinBlockchainBootstrapConfig getBitcoinBlockchainBootstrapConfig() {
+        String blocks = properties.getProperty("BitcoinBlockchainBootstrapConfig.blocks");
+        String firstBlockHeight = properties.getProperty("BitcoinBlockchainBootstrapConfig.firstBlockHeight");
+        if (blocks == null && firstBlockHeight == null)
+            return null;
+
+        if (blocks == null || firstBlockHeight == null)
+            throw new IllegalArgumentException("Must specify either all of none of the BitcoinBlockchainBootstrap config values");
```
Comment (blackyblack) [inline alt-integration-grpc-service/src/main/java/org/veriblock/webservice/DefaultConfiguration.java]: "either all or none"

**Review by blackyblack — APPROVED** [review]

**Comment by Warchant** [comment]
Rebase this PR on develop, so we can merge it

**Pull request merged**